indico/htdocs/js/indico/jquery/categorynavigator.js

```diff
@@ -28,8 +28,19 @@
             openInDialog: false,
             // The title for the category navigator dialog
             dialogTitle: $T.gettext("Select a category"),
-            // Restrict action to categories with no subcategories
-            selectLeafOnly: false,
+            // Allow action to specific categories
+            enableActionOn: {
+                categoriesWithSubcategories: true,
+                categoriesWithEvents: true,
+                self: false
+            },
+            disabledActionMessage: {
+                subcategory: $T.gettext("You cannot select a category that contains subcategories"),
+                event: $T.gettext("You cannot select a category that contains events"),
+                self: $T.gettext("You cannot select the same category")
+            },
+            // The category ID to disable the action for
+            selfCategoryId: null,
```

Comment:
If we go for the nested options this can be specified there directly:

```js
enableActionOn: {
    categoriesWithSubcategories: {
        disabled: true,
        message: $T.gettext("You cannot select a category that contains subcategories")
    },
    categoriesWithEvents: {
        // ...
    }
    self: {
        disabled: false,
        categoryId: null,
        // ...
    }
}
```